scripts/examples_master.py

```diff
@@ -102,3 +102,1212 @@
         },
     ],
 }
+
+EXAMPLES_INFO = {
```

Comment:
This metadata object should leave in its own file, e.g. EXAMPLES_METADATA

I am not sure that a big dict this is the best solution (especially indexed by title...). To start with, it is quite hard to read because the examples are unsorted and all categories are mixed together.

I would suggest something like

```
VISION_EXAMPLES = [
    {
         "title": "Image classification from scratch",
         "path": "...",
         "featured": True,
         "tags": ["tag1", "tag2"],
         "difficulty": 2,
    },
    {
    ...
    },
    ...
```

The ordering of the objects would then become the rendering order on the page.

Reply:
We have child categories like "Image Classification", so we cannot use a flat dict. 

Rethinking about the data structure, I am changing `EXAMPLES_INFO` to a different dict structure - it's indexed by category => sub category => example name, so we have a better way to manage. Please take a look and let me know if you are okay with the design, thanks!

Reply:
> We have child categories like "Image Classification", so we cannot use a flat dict.

Well we could still have the inner objects (per-category) be lists. This is useful because it gives us a manual way to control ordering, which matters (since it's the order in which users read them). Random ordering or alphabetical ordering are not optimal.

Overall I have a few concerns with the current approach:

- Not single-source-of-truth / redundant: we have both a `EXAMPLES_MASTER` list and a separate `EXAMPLES_INFO` dict.
- Not hierarchical. The "category" is a value inside the inner key whereas the subcategory is a key in the outer dict.
- No way to control the display order.

I don't want this issue to be blocking, so I'm happy to merge in the current state and edit afterwards. Do you have any other ideas?

Reply:
The latest push should have resolved 2) and 3). The `EXAMPLES_MASTER ` can be removed theoretically, I will do that with a followup PR. I will address other comments and update the PR, then we can merge it!